astropy/utils/masked/core.py

```diff
@@ -196,7 +196,9 @@ def _get_data_and_mask(cls, data, allow_ma_masked=False):
             try:
                 data = data.unmasked
             except AttributeError:
-                if not isinstance(data, np.ma.MaskedArray):
+                from astropy.nddata import NDDataArray
```

Comment:
Is this absolutely necessary? I would *really* like to remove `NDDataArray` someday...

Reply:
Indeed, would be good to know exactly what goes wrong if this is not here.

If it is really needed, we could try to make this part a bit more duck-typy, assuming that if something has both `.mask` and `.data`, that the latter is the unmasked part. But ideally we do not make any changes here and keep just the backward support for `np.ma.MaskedArray`.